### DISCUSSION TITLE
Working with EEPROM on Arduino Uno

### DISCUSSION THREAD

**Discussion by adhossain** [davetcc/tcMenu#437] [Q&A] [answered]
I was so happy knowing about this framework that I instantly donated few to the developer. But later while working with the code, I saw it is not possible to work on Arduino UNO for most of all displays except 16x2 LCD due to limited space. So I bought 16x2 LCD and tried to add a menu to my project. However, due to limited knowledge, I am not able to save the menu items using EEPROM.

Can anyone give an example of saving and loading the value of a menu from EEPROM for Arduino UNO? @davetcc  your support please.

**Comment by davetcc** [comment, marked as answer]
It depends on which EEPROM you want to use, if you wanted to use the built-in AVR EEPROM or Arduino EEPROM support, it would probably work with Uno, anything using I2C we don't recommend. That's pretty well documented on the site.

Many of the examples include loading and saving from EEPROM, and you can further read this guide: https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/menu-eeprom-integrations/

But honestly, if it is not working for you, feel free to reverse the financial transaction without any hard feelings on our side. Usually with open-source, better to donate after you have tested it works for your case,

**Reply by davetcc** [reply]
Also here's an Uno example that shows how to integrate the support into your project:

https://github.com/davetcc/tcMenuLib/blob/master/examples/avr/analogJoystick1306Ascii/analogJoystick1306Ascii.ino

**Reply by adhossain** [reply]
After playing with the example, I am able to implement the function. Thank you and really appreciate your hard work.

**Comment by adhossain** [comment]
Thank you. I will try and get back to you. Regarding the donation, I would love to be the part of the donation list and I do not have any hesitation regarding this.

Thank  you again
